snuba/web/views.py

```diff
@@ -363,12 +365,36 @@ def sdk_distribution(*, timer: Timer) -> Response:
     return format_result(run_query(dataset, request, timer))
 
 
+SUBSCRIPTION_SEPARATOR = "/"
+
+
+@application.errorhandler(InvalidSubscriptionError)
+def handle_subscription_error(exception: InvalidSubscriptionError):
+    data = {"error": {"type": "subscription", "message": str(exception)}}
+    return (
+        json.dumps(data, indent=4),
+        400,
+        {"Content-Type": "application/json"},
+    )
+
+
+def build_external_subscription_id(identifier: SubscriptionIdentifier) -> str:
+    return "{}{}{}".format(
+        identifier.partition_id, SUBSCRIPTION_SEPARATOR, identifier.subscription_id
+    )
+
+
 @application.route("/<dataset:dataset>/subscriptions", methods=["POST"])
-def create_subscription(*, dataset: Dataset):
-    partition = 0
-    key = uuid1().hex
+@util.time_request("subscription")
+def create_subscription(*, dataset: Dataset, timer: Timer):
+    subscription = SubscriptionCodec().decode(http_request.data)
+    # TODO: Check for valid queries with fields that are invalid for subscriptions. For
+    # example date fields and aggregates.
```

Comment:
Should this happen in `Subscription.__post_init__`?

Reply:
Probably, yeah.